From ea96dbdda3545b83bb9408449a5c625475df84bc Mon Sep 17 00:00:00 2001
From: Xulin Sun <xulin.sun@windriver.com>
Date: Wed, 25 Apr 2018 13:18:32 +0800
Subject: [PATCH 3/5] ARM: dts: ls1021a: remove dma-coherent for IOT board

There is a HW defect in ls1021a IOT board, it is known that there are
H/W coherency issues at least for the Tx side. So remove the dma-coherent
for IOT board like SDK
Freescale-Linux-SDK-for-LS1021A-IOT-Rev2-v0.4-20150907-yocto.iso has done.

Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 arch/arm/boot/dts/ls1021a-twr.dts | 3 +++
 arch/arm/boot/dts/ls1021a.dtsi    | 3 ---
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/arch/arm/boot/dts/ls1021a-twr.dts b/arch/arm/boot/dts/ls1021a-twr.dts
index 760d18d..c74e85c 100644
--- a/arch/arm/boot/dts/ls1021a-twr.dts
+++ b/arch/arm/boot/dts/ls1021a-twr.dts
@@ -159,6 +159,7 @@
 	tbi-handle = <&tbi1>;
 	phy-handle = <&sgmii_phy2>;
 	phy-connection-type = "sgmii";
+	dma-coherent;
 	status = "okay";
 };
 
@@ -166,12 +167,14 @@
 	tbi-handle = <&tbi1>;
 	phy-handle = <&sgmii_phy0>;
 	phy-connection-type = "sgmii";
+	dma-coherent;
 	status = "okay";
 };
 
 &enet2 {
 	phy-handle = <&rgmii_phy1>;
 	phy-connection-type = "rgmii-id";
+	dma-coherent;
 	status = "okay";
 };
 
diff --git a/arch/arm/boot/dts/ls1021a.dtsi b/arch/arm/boot/dts/ls1021a.dtsi
index 653276d..4523c9b 100644
--- a/arch/arm/boot/dts/ls1021a.dtsi
+++ b/arch/arm/boot/dts/ls1021a.dtsi
@@ -630,7 +630,6 @@
 			model = "eTSEC";
 			fsl,magic-packet;
 			ranges;
-			dma-coherent;
 
 			queue-group@2d10000 {
 				#address-cells = <2>;
@@ -659,7 +658,6 @@
 			interrupt-parent = <&gic>;
 			model = "eTSEC";
 			ranges;
-			dma-coherent;
 
 			queue-group@2d50000  {
 				#address-cells = <2>;
@@ -688,7 +686,6 @@
 			interrupt-parent = <&gic>;
 			model = "eTSEC";
 			ranges;
-			dma-coherent;
 
 			queue-group@2d90000  {
 				#address-cells = <2>;
-- 
1.8.5.2.233.g932f7e4

